subprojects/diagnostics/src/main/java/org/gradle/api/reporting/model/internal/ModelNodeRenderer.java

```diff
@@ -163,12 +163,7 @@ private String attributeLabel(String label) {
     }
 
     static Iterable<ModelRuleDescriptor> uniqueExecutedRulesExcludingCreator(final ModelNode model) {
-        Iterable filtered = Iterables.filter(model.getExecutedRules(), new Predicate<ModelRuleDescriptor>() {
-            @Override
-            public boolean apply(ModelRuleDescriptor input) {
-                return !input.equals(model.getDescriptor());
-            }
-        });
+        Iterable<ModelRuleDescriptor> filtered = model.getExecutedRules().stream().filter(input -> !input.equals(model.getDescriptor())).collect(Collectors.toList());
```

Comment:
Could have used the collector `ImmutableSet#toImmutableSet` right away

Reply:
Yes, I searched for it ... not well enough - fixing

Reply:
Ah, I know why I missed it: available only in the `jre` variant, but we still use the `android` one because of a perf regression: https://github.com/google/guava/issues/3223